moped-editor/src/views/projects/projectView/ProjectComments.js

```diff
@@ -214,11 +210,8 @@ const ProjectComments = props => {
       });
     }
     refetch();
-  }, [noteType, setNoteTypeConditions, refetch]);
+  }, [noteType, setNoteTypeConditions, refetch, data]);
```

Comment:
we talked about potentially putting a more specific variable `data?.moped_proj_notes` here instead of `data`. I don't really have a preference but kinda curious if that would work too.

Reply:
so I did move it out into its own separate variable, instead of just checking all of data

Reply:
hmm, i'm kinda surprised this isn't causing an infinite hook loop, since I would expect `data` to change after `refetch()`. i wonder if  apollo caching is keeping the data object reference persistent and preventing infinity... but in any case this doesn't look ideal to me.

running locally, i noticed we're firing off the same graphql query three times when you toggle between note types. i believe those are:

- `useQuery` fires after `filterNoteType()` changes the query params
- `refetch()` fires in `useEffect` b/c of `noteType` dependency
- `refetch()` fires again in `useEffect` because the `data` dependency changed

haven't debugged much, but yeah i'm stumped as to why this bug is happening at all.

Reply:
could the `setTimeout`s be messing up state in an unexpected way? we use those just to force the animations to persist for a sec?